src/org/jetbrains/java/decompiler/struct/gen/MethodDescriptor.java

```diff
@@ -90,8 +90,15 @@ public static MethodDescriptor parseDescriptor(StructMethod struct, ClassNode no
         }
         else if (init) {
           // && isEnum
-          if (DecompilerContext.getOption(IFernflowerPreferences.DECOMPILE_ENUM) && DecompilerContext.getStructContext().getClass(struct.getClassQualifiedName()).hasModifier(CodeConstants.ACC_ENUM)) {
-            actualParams -= 2;
+          if (DecompilerContext.getOption(IFernflowerPreferences.DECOMPILE_ENUM)) {
+            String className = struct.getClassQualifiedName();
+            if (DecompilerContext.getOption(IFernflowerPreferences.RENAME_ENTITIES) && (className = DecompilerContext.getPoolInterceptor().getOldName(className)) == null) {
```

Comment:
The inline assignment to `className`  should be replaced, as I think it hurts code readability. It'd be better to use a compound if statement here.

Reply:
I assume my new commit is what you mean